src/langchain_google_spanner/vector_store.py

```diff
@@ -985,9 +1305,26 @@ def similarity_search_by_vector(
         Returns:
             List[Document]: List of documents most similar to the query.
         """
-        documents = self.similarity_search_with_score_by_vector(
-            embedding=embedding, k=k, pre_filter=pre_filter
-        )
+        documents: List[Document] = None
+        if (
+            self._query_parameters.algorithm
+            == QueryParameters.NearestNeighborsAlgorithm.APPROXIMATE_NEAREST_NEIGHBOR
+        ):
+            documents = self.__search_by_ANN(
+                index_name=kwargs.get("index_name", None),
+                num_leaves=kwargs.get("num_leaves", 1000),
+                k=k,
+                embedding=embedding,
+                embedding_column_is_nullable=kwargs.get("embedding_column_is_nullable", False),
+                pre_filter=kwargs.get("pre_filter", ""),
```

Comment:
You can also define it as Optional Parameter in the function signature... so that customer knows these parameters are available.

Reply:
It was defined in the signature for similary_search_by_vector.